### PR TITLE
fix: only load newer macOS sdks

### PR DESCRIPTION
**Pull request by codebytere**
We should only be linking 10.14 & 10.15, as older ones now cause compilation errors.

cc @MarshallOfSound

### CODE REVIEW

**Comment by MarshallOfSound** [comment]
Can we add a thing to unlink 10.13

File: src/e-load-xcode.js [hunk 1]
```diff
@@ -48,6 +51,23 @@ for (const sdk of SDK_TO_LINK) {
   childProcess.execFileSync('sudo', ['ln', '-s', sourceSDK, targetDirectory]);
 }
 
+// Unlink unnecessary macOS SDKs.
+for (const sdk of SDK_TO_UNLINK) {
+  const sourceSDK = path.resolve(macOSSDKs.path, `MacOSX${sdk}.sdk`);
+  if (!fs.existsSync(sourceSDK)) continue;
+
+  const targetDirectory = path.resolve(xCodeSDKDir, `MacOSX${sdk}.sdk`);
+  if (!fs.existsSync(targetDirectory)) continue;
```
Comment (MarshallOfSound) [inline src/e-load-xcode.js]: Can we check that the target is actually a symbolic link, some version of `fs.statSync()` should work

File: src/e-load-xcode.js [hunk 2]
```diff
@@ -48,6 +51,23 @@ for (const sdk of SDK_TO_LINK) {
...
+
+  console.warn(
+    `${color.info} Removing symbolic link from ${color.path(sourceSDK)} --> ${color.path(
+      targetDirectory,
+    )}`,
+  );
+
+  childProcess.execFileSync('sudo', ['unlink', sourceSDK]);
```
Comment (MarshallOfSound) [inline src/e-load-xcode.js]: I've just realized I don't think we need `sudo` for this any more but I'll check in a separate PR

Comment (MarshallOfSound) [reply src/e-load-xcode.js]: Yeah we don't need `sudo` for this https://github.com/electron/build-tools/pull/112/files

File: src/e-load-xcode.js [hunk 3]
```diff
@@ -48,6 +53,29 @@ for (const sdk of SDK_TO_LINK) {
...
+  const stats = fs.lstatSync(targetDirectory);
+  if (!stats.isSymbolicLink()) return;
+
+  console.warn(
+    `${color.info} Removing symbolic link from ${color.path(sourceSDK)} --> ${color.path(
+      targetDirectory,
+    )}`,
+  );
+
+  childProcess.execFileSync('unlink', [sourceSDK]);
```
Comment (MarshallOfSound) [inline src/e-load-xcode.js]: Shouldn't this be `unlink', [targetDirectory])` ?

Comment (codebytere) [reply src/e-load-xcode.js]: derp, ye. fixed.

**Pull request merged**